openapi-cli/src/main/java/io/ballerina/generators/client/FunctionSignatureGenerator.java

```diff
@@ -350,9 +353,20 @@ public Node getPathParameters(Parameter parameter) throws BallerinaOpenApiExcept
         NodeList<AnnotationNode> annotationNodes =
                 docCommentsGenerator.extractDisplayAnnotation(parameter.getExtensions());
         IdentifierToken paramName = createIdentifierToken(escapeIdentifier(parameter.getName().trim()));
-        String type = convertOpenAPITypeToBallerina(parameter.getSchema().getType().trim());
-        if (type.equals("anydata") || type.equals("[]") || type.equals("record {}")) {
-            throw new BallerinaOpenApiException(invalidPathParamType(parameter.getName().trim()));
+        String type = "";
+        Schema parameterSchema = parameter.getSchema();
+        if (parameterSchema.get$ref() != null) {
+            type = getValidName(extractReferenceType(parameterSchema.get$ref()), true);
+            Schema schema = openAPI.getComponents().getSchemas().get(type.trim());
+            if (schema instanceof ObjectSchema) {
```

Comment:
Does the OpenAPI allow to have object schema in Path parameter?

Reply:
Yes, I have added a test case for this scenario [here](https://github.com/ballerina-platform/ballerina-openapi/blob/be005211def91caf8b1f8e46f12be42b00592c77/openapi-cli/src/test/java/io/ballerina/generators/client/PathParameterTests.java#L63)

Reply:
it is fine :)